rust/src/treefile.rs

```diff
@@ -271,8 +271,10 @@ fn merge_basic_field<T>(dest: &mut Option<T>, src: &mut Option<T>) {
 /// the `packages` key.
 fn merge_vec_field<T>(dest: &mut Option<Vec<T>>, src: &mut Option<Vec<T>>) {
     if let Some(mut srcv) = src.take() {
-        if let Some(ref mut destv) = dest {
-            srcv.append(destv);
+        if let Some(mut destv) = dest.take() {
```

Comment:
The next patches will effectively add a unit test for this right?  If so fine as is but if not it'd be good to add one.

Reply:
Yes, https://github.com/coreos/rpm-ostree/pull/2787 has tests which rely on this.

Reply:
Ohhh wait, yes I think the postprocess scripts are relying on the current ordering.